mmcls/models/backbones/levit.py

```diff
@@ -0,0 +1,545 @@
+    @torch.no_grad()
+    def train(self, mode=True):
+        super(Attention, self).train(mode)
+        if mode and hasattr(self, 'ab'):
+            del self.ab
+        else:
+            self.ab = self.attention_biases[:, self.attention_bias_idxs]
+
+    def forward(self, x):  # x (B,N,C)
+        B, N, C = x.shape  # 2 196 128
+        qkv = self.qkv(x)  # 2 196 128
+        q, k, v = qkv.view(B, N, self.num_heads, -1).split(
+            [self.key_dim, self.key_dim, self.d],
+            dim=3)  # q 2 196 4 16 ; k 2 196 4 16; v 2 196 4 32
+        q = q.permute(0, 2, 1, 3)  # 2 4 196 16
+        k = k.permute(0, 2, 1, 3)
+        v = v.permute(0, 2, 1, 3)
+
+        attn = ((q @ k.transpose(-2, -1)) *
+                self.scale  # 2 4 196 16 * 2 4 16 196 -> 2 4 196 196
+                + (self.attention_biases[:, self.attention_bias_idxs]
+                   if self.training else self.ab))
+        attn = attn.softmax(dim=-1)  # 2 4 196 196 -> 2 4 196 196
+        x = (attn @ v).transpose(1, 2).reshape(
+            B, N,
+            self.dh)  # 2 4 196 196 * 2 4 196 32 -> 2 4 196 32 -> 2 196 128
+        x = self.proj(x)
+        return x
+
+
+class MLP(nn.Sequential):
+
+    def __init__(
+        self,
+        embed_dim,
+        mlp_ratio,
+        mlp_activation,
+    ):
+        super(MLP, self).__init__()
+        h = embed_dim * mlp_ratio
+        self.linear1 = LinearBatchNorm(embed_dim, h)
+        self.activation = mlp_activation()
+        self.linear2 = LinearBatchNorm(h, embed_dim, bn_weight_init=0)
+
+    def forward(self, x):
+        x = self.linear1(x)
+        x = self.activation(x)
+        x = self.linear2(x)
+        return x
+
+
+class Subsample(BaseModule):
+
+    def __init__(self, stride, resolution):
+        super(Subsample, self).__init__()
+        self.stride = stride
+        self.resolution = resolution
+
+    def forward(self, x):
+        B, N, C = x.shape  # 2 196 128
+        x = x.view(B, self.resolution, self.resolution,
+                   C)[  # 2 196 128 -> 2 14 14 128  -> 2 7 7 128
+                       :, ::self.stride, ::self.stride].reshape(
+                           B, -1, C)  # # 2 14 14 128 > 2 49 128
+        return x
+
+
+class AttentionSubsample(nn.Sequential):
+
+    def __init__(self,
+                 in_dim,
+                 out_dim,
+                 key_dim,
+                 num_heads=8,
+                 attn_ratio=2,
+                 activation=None,
+                 stride=2,
+                 resolution=14,
+                 resolution_=7):
+        super(AttentionSubsample, self).__init__()
+        self.num_heads = num_heads
+        self.scale = key_dim**-0.5
+        self.key_dim = key_dim
+        self.nh_kd = nh_kd = key_dim * num_heads
+        self.d = int(attn_ratio * key_dim)
+        self.dh = int(attn_ratio * key_dim) * self.num_heads
+        self.attn_ratio = attn_ratio
+        self.resolution_ = resolution_
+        self.resolution_2 = resolution_**2
```

Comment:
Why do we need to calculate a square here instead of calculating it when using it?